devmgmtV2/controllers/auth.controller.js

```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
   return selectFields(queryObject);
 }
 
-/*
- * Custom method to convert a stringified list into a normal one, because JSON.parse is throwing a fit
- */
-
-function listify(string) {
-  let lst = [];
-  string = string.slice(1,-1);
-  lst = string.split(',');
-  for (var i in lst) {
-    lst[i] = lst[i].trim().slice(1, -1);
-  }
-  return lst;
-}
-
-/*
- * The authLogin and updatePW return three values:
- *     1. true if the operation is successful
- *     2. false if the user does not exist
- *     3. null if there is a MYSQL error [happens during SQL Injection attacks]
- * The getPerms function returns either the list of permission [if a valid user exists],
- * null otherwise.
- */
+//SQL injection attempts lead to syntax errors in the SQL statements, and are handled with 500s,
 
 let authLogin = (req,res) => {
   let userName = req.body['uname'];
   let password = req.body['pword'];
-  let result = true;
   verifyIfUserExists(userName).then(response => {
-    if (typeof response[0] === "undefined" || response[0]['password'] != password) {
-      result = false;
-    }
-    res.json({'isSuccessful' : result});
-    }, error => {
-      console.log(error);
-      res.json({'isSuccessful' : null});
+    return new Promise((resolve, reject) => {
+      if (response[0] === "undefined" || response[0]['password'] != password) {
+        reject(false);
+      }
+      else {
+        resolve(getColumnFromDB(userName, 'permission'));
+      }
     });
+  }, reject => {
+    console.log(reject);
+    res.status(500).json({authSuccessful : false, authMessage : "Internal server error! Possible SQL Injection attempt"});
+  }).then(response => {
+    let permissionList = JSON.parse(response[0]['permission']);
```

Comment:
front end will do it, you don't handle this, if this break ... your server will break